rubocop-packaging.gemspec

```diff
@@ -23,10 +23,10 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = Gem::Requirement.new('>= 2.3.0')
 
-  spec.add_development_dependency   'bump', '>= 0.8.0'
-  spec.add_development_dependency   'pry', '>= 0.13.0'
+  spec.add_development_dependency   'bump', '~> 0.8'
+  spec.add_development_dependency   'pry', '~> 0.13'
   spec.add_development_dependency   'rake', '~> 13.0'
   spec.add_development_dependency   'rspec', '~> 3.0'
   spec.add_development_dependency   'yard', '~> 0.9'
-  spec.add_runtime_dependency       'rubocop', '>= 0.75.0'
+  spec.add_runtime_dependency       'rubocop', '~> 0.86.0'
```

Comment:
Sayy whaaaaat!?
So the specs fail with RuboCop v0.88.0 but passes with RuboCop v0.86.0. Anything after v0.86.0 just doesn't work. D'oh!?

Reply:
Perhaps, might relate to https://github.com/rubocop-hq/rubocop/pull/7868.
(mostly because anything after 0.86 just doesn't work)

And according to RuboCop's [CHANGELOG](https://github.com/rubocop-hq/rubocop/blob/master/CHANGELOG.md), the 3 things there are marked as **breaking**.

Reply:
Ughhhh.
https://docs.rubocop.org/rubocop/0.87/v1_upgrade_notes.html#detailed-api-changes

Reply:
cf: https://docs.rubocop.org/rubocop/0.87/v1_upgrade_notes.html

Reply:
I am trying to figure this out with the following diff:
```
--- a/config/default.yml
+++ b/config/default.yml
@@ -8,4 +8,4 @@ Packaging/GemspecGit:
 Packaging/RelativeRequireToLib:
   Description: 'Avoid using `require` with relative path to lib.'
   Enabled: true
-  VersionAdded: '0.86'
+  VersionAdded: '0.89'
diff --git a/lib/rubocop/cop/packaging/relative_require_to_lib.rb b/lib/rubocop/cop/packaging/relative_require_to_lib.rb
index a766b59..c6761f3 100644
--- a/lib/rubocop/cop/packaging/relative_require_to_lib.rb
+++ b/lib/rubocop/cop/packaging/relative_require_to_lib.rb
@@ -8,7 +8,7 @@ module RuboCop # :nodoc:
       # Write bad and good methods.
       # Help from cop/lint/redundant_require_statement.rb.
       #
-      class RelativeRequireToLib < Cop
+      class RelativeRequireToLib < Base
         # This is the message that will be displayed when RuboCop finds an
         # offense of using `require` with relative path to lib.
         MSG = 'Avoid using `require` with relative path to lib.'
@@ -18,7 +18,7 @@ module RuboCop # :nodoc:
             (str #starts_with_relative_path?))
         PATTERN
 
-        def investigate(processed_source)
+        def on_new_investigation
           @file_path = processed_source.file_path
           @file_directory = File.dirname(@file_path)
         end
@@ -28,7 +28,7 @@ module RuboCop # :nodoc:
         def on_send(node)
           return unless require_relative(node)
 
-          add_offense(node)
+          add_offense(node.loc.selector, message: MSG)
         end
 
```

...but to no avail.

Reply:
Hm, looks like I am missing something, not sure what :disappointed:

Reply:
I think you're on the right track :muscle:!

Reply:
`add_offense(node)` can remain the same, no problem. `on_new_investigation` is good

Reply:
Wow @marcandre, thanks so much for throwing some feedback in here, that's super nice of you :heart: .

All of your work in `rubocop` is amazing by the way :)